### PR TITLE
RFC 82: Support for Enhanced Layer Metadata Management

### PR DESCRIPTION
**Pull request by tomkralidis**
Per http://mapserver.org/development/rfc/ms-rfc-82.html

### CODE REVIEW

**Comment by tomkralidis** [comment]
FYI this PR is ready for review.  Notes:

- implements layer level metadata. Service level metadata not implemented as part of first pass
- Travis errors are either unrelated or I'm not able to reproduce them locally
- there are no backward compatibility issues, however the implementation results in WMS/WFS/WCS Capabilities **always** providing `MetadataURL` type elements

**Comment by tbonfort** [comment]
TRavis errors are due to a ```&amp;amp;``` appearing in generated capabilities, needs fixing

File: mapmetadata.c [hunk 1]
```diff
@@ -0,0 +1,900 @@
...
+    temporal = msStringSplit(value, '/', &n);
+  }
+  else { /* WCS */
+    value = (char *)msOWSLookupMetadata(&(layer->metadata), "CO", "timeposition");
+    if (value) {
+      /* split extent */
+      temporal = msStringSplit(value, ',', &n);
+    }
+  }
+  if (!value) /* SOS */
```
Comment (tbonfort) [inline mapmetadata.c]: wrong bracketing or indentation

**Comment by tomkralidis** [comment]
@tbonfort thanks fixed. Travis working now.  Commits squashed.

**Comment by tomkralidis** [comment]
Will update docs this week.

**Comment by tomkralidis** [comment]
@jmckenna given this is new functionality targeted for 7.2, should docs PR be written against master?

**Comment by jmckenna** [comment]
@tomkralidis yes agreed, sorry for the confusion.

**Pull request merged**